domain/application/service/storage.go

```diff
@@ -22,6 +22,17 @@ import (
 type StorageDirectiveOverrides = storage.StorageDirectiveOverride
 
 type StorageService interface {
+	// GetApplicationStorageDirectivesInfo returns the storage directives set for an application,
+	// keyed to the storage name. If the application does not have any storage
+	// directives set then an empty result is returned.
+	//
+	// If the application does not exist, then a [applicationerrors.ApplicationNotFound]
```

Comment:
do we document the expected error in the interface definition? i would expect it to be documented in the implementation.

Reply:
As discussed, I just noticed it seems to be a convention